src/libs/ReportActionsUtils.ts

```diff
@@ -635,6 +635,39 @@ function isReportActionDeprecated(reportAction: OnyxEntry<ReportAction>, key: st
     return false;
 }
 
+/**
+ * Checks if a given report action corresponds to an actionable mention whisper.
+ * @param reportAction
+ */
+function isActionableMentionWhisper(reportAction: OnyxEntry<ReportAction>): reportAction is ReportAction<typeof CONST.REPORT.ACTIONS.TYPE.ACTIONABLE_MENTION_WHISPER> {
+    return isActionOfType(reportAction, CONST.REPORT.ACTIONS.TYPE.ACTIONABLE_MENTION_WHISPER);
+}
+
+/**
+ * Checks if a given report action corresponds to an actionable report mention whisper.
+ * @param reportAction
+ */
+function isActionableReportMentionWhisper(reportAction: OnyxEntry<ReportAction>): reportAction is ReportAction<typeof CONST.REPORT.ACTIONS.TYPE.ACTIONABLE_REPORT_MENTION_WHISPER> {
+    return isActionOfType(reportAction, CONST.REPORT.ACTIONS.TYPE.ACTIONABLE_REPORT_MENTION_WHISPER);
+}
+
+/**
+ * Checks whether an action is actionable track expense.
+ */
+function isActionableTrackExpense(reportAction: OnyxInputOrEntry<ReportAction>): reportAction is ReportAction<typeof CONST.REPORT.ACTIONS.TYPE.ACTIONABLE_TRACK_EXPENSE_WHISPER> {
+    return isActionOfType(reportAction, CONST.REPORT.ACTIONS.TYPE.ACTIONABLE_TRACK_EXPENSE_WHISPER);
+}
+
+function isActionableWhisper(
+    reportAction: OnyxEntry<ReportAction>,
+): reportAction is ReportAction<
```

Comment:
NAB - for this one and the ones similar to it... How does this return type work? It's just explaining that the function will return whether or not the reportAction is of those types? Is it so that other parts of the code can know that the reportAction is/isn't those types, outside of our actual logic that says that?

I think it makes sense, it's just hard to wrap my brain around the Typescript of it.

Reply:
> https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

Yes. This is used for a generic type. For example, `isActionableTrackExpense` will return a boolean whether `reportAction` is of type `ReportAction<typeof CONST.REPORT.ACTIONS.TYPE.ACTIONABLE_TRACK_EXPENSE_WHISPER>` or not. If we return `true`, then the report action inside the `if` block will have a type of `ReportAction<typeof CONST.REPORT.ACTIONS.TYPE.ACTIONABLE_TRACK_EXPENSE_WHISPER>`.

here is the ts doc for it: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates